### PR TITLE
8301095: [s390x] TestDwarf.java fails

### PR DESCRIPTION
**Pull request by offamitkumar**
TestDwarf.java fails on s390 due to incorrect returned frame. for dereference_null crash, the (return) pc was selected from gpr14 (in accordance with z-ABI) but correct (return) pc was present at "return_pc" offset. This PR introduces an additional check for the frame type detection. 
Special Thanks to @RealLucy

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8301095](https://bugs.openjdk.org/browse/JDK-8301095): [s390x] TestDwarf.java fails

### Reviewers
 * [Lutz Schmidt](https://openjdk.org/census#lucy) (@RealLucy - **Reviewer**)
 * [Tyler Steele](https://openjdk.org/census#tsteele) (@backwaterred - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12306/head:pull/12306` \
`$ git checkout pull/12306`

Update a local copy of the PR: \
`$ git checkout pull/12306` \
`$ git pull https://git.openjdk.org/jdk pull/12306/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12306`

View PR using the GUI difftool: \
`$ git pr show -t 12306`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12306.diff">https://git.openjdk.org/jdk/pull/12306.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/os_cpu/linux_s390/os_linux_s390.cpp [hunk 1]
```diff
@@ -172,7 +172,14 @@ frame os::get_sender_for_C_frame(frame* fr) {
         return frame(fr->sender_sp(), fr->sender_pc());
       }
     } else {
-      return frame(fr->sender_sp(), fr->native_sender_pc());
+        intptr_t* sender_sp = fr->sender_sp();
+        address   sender_fp = (address)*sender_sp;
+        ptrdiff_t entery_len = sender_fp - (address)sender_sp;
```
Comment (RealLucy) [inline src/hotspot/os_cpu/linux_s390/os_linux_s390.cpp:177]: I guess this should read "entry_len".

File: src/hotspot/os_cpu/linux_s390/os_linux_s390.cpp [hunk 2]
```diff
@@ -172,7 +172,14 @@ frame os::get_sender_for_C_frame(frame* fr) {
         return frame(fr->sender_sp(), fr->sender_pc());
       }
     } else {
-      return frame(fr->sender_sp(), fr->native_sender_pc());
```
Comment (RealLucy) [inline src/hotspot/os_cpu/linux_s390/os_linux_s390.cpp:175]: We use 2-space indentation for c/cpp files.

**Comment by offamitkumar** [comment]
Hi @RealLucy, Thanks for review. 

I've made some changes accordingly.

**Review by RealLucy — APPROVED** [review]
LGTM

**Review by backwaterred — APPROVED** [review]

**Comment by backwaterred** [comment]
These changes LGTM also.

/sponsor

**Comment by RealLucy** [comment]
@offamitkumar I'm sorry, we're not there yet. HotSpot changes require two reviews before they can be integrated. Reviewers for s390 PRs are a scarce resource. I've already pinged @TheRealMDoerr to help with this PR.

**Comment by backwaterred** [comment]
Sorry for stepping on your toes @RealLucy. I am happy to receive any feedback from Martin as well.

While I don't normally review s390 changes, I felt familiar enough with these changes to give it a +1.

**Comment by RealLucy** [comment]
Ahhh, no problem at all. I didn't feel any pain.

**Pull request closed without merging**